api/v1alpha1/loadbalancer_types.go

```diff
@@ -52,9 +52,27 @@ const (
 )
 
 // ConsistentHash defines the configuration related to the consistent hash
-// load balancer policy
+// load balancer policy.
+// +union
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'Header' ? has(self.header) : !has(self.header)",message="If consistent hash type is header, the header field must to be set."
 type ConsistentHash struct {
+	// Valid Type values are
+	// "SourceIP",
+	// "Header",
+	//
+	// +unionDiscriminator
 	Type ConsistentHashType `json:"type"`
+
+	// Header configures the header hash policy when the consistent hash type is set to Header.
+	// +optional
+	Header *Header `json:"header,omitempty"`
+}
+
+// Header defines the header hashing configuration for consistent hash based
+// load balancing.
+type Header struct {
+	HeaderName string `json:"header_name"`
```

Comment:
following convention used in other place, just `Name` makes more sense here
* https://github.com/envoyproxy/gateway/blob/ed4222b78124de25d0c7094080567c45047d33c3/api/v1alpha1/ratelimit_types.go#L144
* https://github.com/kubernetes-sigs/gateway-api/blob/a29ddd4ed9395c0c192c19be853deb5e0a3a854d/apis/v1/httproute_types.go#L469

Reply:
Renamed!